io/io/test/TFileMergerTests.cxx

```diff
@@ -219,3 +219,39 @@ TEST(TFileMerger, ChangeFile)
    gSystem->Unlink("file6640mergerinput_2.root");
    gSystem->Unlink("file6640mergeroutput.root");
 }
+
+// https://github.com/root-project/root/issues/9022
+TEST(TFileMerger, SingleHistFile)
+{
+   auto filename1 = "f1_9022.root";
+   auto filename2 = "f2_9022.root";
+   auto outname = "file9022mergeroutput.root";
+   {
+      TFile f1(filename1, "RECREATE");
+      TH1F h("h1", "h1", 1, 0, 1);
+      h.Write();
+      f1.Close();
+      TFile f2(filename2, "RECREATE");
+      TH1F h2("h2", "h2", 1, 0, 1);
+      h2.Write();
+      f2.Close();
+   }
+   {
+      TFileMerger filemerger{false, false};
+      filemerger.SetMaxOpenedFiles(2);
+      filemerger.OutputFile(std::unique_ptr<TFile>{TFile::Open(outname, "RECREATE")});
+
+      filemerger.AddFile(filename1);
+      filemerger.AddFile(filename2);
+
+      filemerger.Merge();
+   }
+   {
+      TFile file(outname, "READ");
+      EXPECT_NE(file.Get<TH1>("h1"), nullptr);
+      EXPECT_NE(file.Get<TH1>("h2"), nullptr);
+   }
+   gSystem->Unlink(filename1);
+   gSystem->Unlink(filename2);
+   gSystem->Unlink(outname);
+}
```

Comment:
Hello @ferdymercury,

you seem to be working on a similar problem (I have some changes in #19331), so maybe you could cherry-pick the following test to see if your PR fixes it:
https://github.com/root-project/root/commit/46148c75da52151861e92e17325113cb81bc6787
[branch](https://github.com/hageboeck/root/tree/TFileMerger_mergeSelective)